paddle/operators/math/math_function.cu

```diff
@@ -135,16 +135,16 @@ void matmul<platform::GPUPlace, double>(
   auto dim_a = matrix_a.dims();
   auto dim_b = matrix_b.dims();
   auto dim_out = matrix_out->dims();
-  PADDLE_ENFORCE(dim_a.size() == 2 && dim_b.size() == 2 && dim_out.size() == 2,
```

Comment:
It's reasonable to restrict the output tensor to rank 2. I think we shouldn't remove this check.